src/redux/sagas/onboarding.js

```diff
@@ -0,0 +1,51 @@
+import {
+  put, takeLatest, call, select,
+} from 'redux-saga/effects';
+
+import { blockchainActions, onBoardingActions } from '../actions';
+import { blockchainSelectors } from '../selectors';
+import { getComplimentaryLLD } from '../../api/backend';
+import { getBalanceByAddress } from '../../api/nodeRpcCall';
+
+// WORKERS
+
+function* claimComplimentaryLLDWorker() {
+  try {
+    const walletAddress = yield select(blockchainSelectors.userWalletAddressSelector);
+    const getComplimentaryLLDResponse = yield call(getComplimentaryLLD, walletAddress);
+    let errorData = {};
+    if (getComplimentaryLLDResponse.status === 200) {
+      // FIXME eventually we need a notification modal, not just error modal
+      errorData = { details: 'Complimentary LLDs claimed. Please refresh the page' };
+      yield put(blockchainActions.setErrorExistsAndUnacknowledgedByUser.success(true));
+      yield put(blockchainActions.setError.success(errorData));
+      yield call(getBalanceByAddress, walletAddress);
+    } else {
+      errorData = {
+        details: typeof getComplimentaryLLDResponse?.data === 'string'
+          ? getComplimentaryLLDResponse?.data
+          : 'Technical error without description, please let the devs know',
+      };
+      yield put(blockchainActions.setErrorExistsAndUnacknowledgedByUser.success(true));
+      yield put(blockchainActions.setError.success(errorData));
+    }
+  } catch {
+    const errorData = { details: 'Error in claiming, send screenshot to devs please' };
+    yield put(blockchainActions.setErrorExistsAndUnacknowledgedByUser.success(true));
+    yield put(blockchainActions.setError.success(errorData));
```

Comment:
I dont understand what do you mean by implementing blockchainWatcher?

 (and i dont think implementing complexity to save few lines is worth it)

Reply:
> I dont understand what do you mean by implementing blockchainWatcher?

Something like this:

```javascript
function* claimComplimentaryLLDWorker() {
  const walletAddress = yield select(blockchainSelectors.userWalletAddressSelector);
  const getComplimentaryLLDResponse = yield call(getComplimentaryLLD, walletAddress);
  let errorData = {};
  if (getComplimentaryLLDResponse.status === 200) {
    yield put(onBoardingActions.getEligibleForComplimentaryLld.call());
    yield put(onBoardingActions.claimComplimentaryLld.success());
    yield call(getBalanceByAddress, walletAddress);
  } else {
    errorData = {
      details: typeof getComplimentaryLLDResponse?.data === 'string'
        ? getComplimentaryLLDResponse?.data
        : 'Technical error without description, please let the devs know',
    };
    throw new Error({errorData});
  }
}

function* claimComplimentaryLLDWatcher() {
  yield* blockchainWatcher(onBoardingActions.claimComplimentaryLld, claimComplimentaryLLDWorker);
}
```
> unless its a solution common through the codebase

It is used in all new code already. For example, most of the congress: https://github.com/liberland/liberland_frontend/blob/master/src/redux/sagas/congress.js#L168-L343

> and i dont think implementing complexity

IMO it reduces complexity. Now instead of manual error handling in each case you don't have to do anything in most cases and only throw error in special cases.

> to save few lines

It's not just a few lines. It's ~8-10 lines for each action handler. Given we have dozens of actions, this results in hundreds of lines. It's easily 30% of code in our sagas. And historically those lines were ridden with bugs - eating up errors completely, not passing correct objects to actions, forgetting some actions (like here - in all this nested ifs it doesn't emit `success` action although it's defined and probably needed to handle loaders correctly).